src/posix/main.c

```diff
@@ -261,10 +261,11 @@ static otInstance *InitInstance(PosixConfig *aConfig)
 
     syslog(LOG_INFO, "Running %s", otGetVersionString());
     syslog(LOG_INFO, "Thread version: %hu", otThreadGetVersion());
-    IgnoreError(otLoggingSetLevel(aConfig->mLogLevel));
 
     instance = otSysInit(&aConfig->mPlatformConfig);
 
+    IgnoreError(otLoggingSetLevel(aConfig->mLogLevel));
+
     atexit(otSysDeinit);
```

Comment:
This fixes a bug where otSysInit() sets the log level to the compile time config, rendering the command line argument useless.